commands/command_edit.cpp

```diff
@@ -305,11 +304,11 @@ QString EditAtmPress::fieldName() const
 // ***** Duration *****
 void EditDuration::set(struct dive *d, int value) const
 {
-	d->dc.duration.seconds = value;
-	d->duration = d->dc.duration;
-	d->dc.meandepth.mm = 0;
-	d->dc.samples = 0;
-	fake_dc(&d->dc);
+	d->dcs[0].duration.seconds = value;
```

Comment:
We can only set the duration on "manually added" dives. Not sure what happens if a manually added dive is merged with a dive computer dive. That should be checked. But as noted above: I tried to avoid changing behavior.

Reply:
We are definitely not doing anything to stop users from combining manually added dives with other dives. And there might be a use case for it when people use manually added dives as a 'poor man's dive planner' for recreational dives.
And as I've found out in #4181, #4205, and #4210, there are A LOT of places where our code blindly uses or modifies the first dive computer in a dive, and ends up creating a mess in multi dive computer dives.
So this is probably an area where a lot of buggy behaviour can by fixed by addressing this one thing.